### PR TITLE
Better switch handling for CleanRepo tool

### PR DESCRIPTION
**Pull request by IEvangelist**
## Background

I can call a bash script from the _Dockerfile_, however, I cannot access the command line args given to the docker run command, from within the _Dockferfile_ itself. In looking at the clean repo tool exe, I'm proposing that we change the `if / else if / else` in the `RunOptions` with standalone `if` statements. This means, you can give your tool all the options you want it to perform, and it will do all of them in sequence. I believe that this is a non-breaking change, but at least allows us the flexibility to call the tool once if we eventually desire that.

**TL;DR;**

The problem is that as a GitHub Action, in order to get the arguments passed to it from a consuming workflow, we must use the existing GitHub Action infrastructure. It passes their arguments as command line args to the `docker run` command, as shown here in this example (an example Action run from the quest importer which uses the same docker GHA model):

```shell
/usr/bin/docker run --name ccc55a73283a34d22a3445df6d27b1f47_b30ba5 
  --label 49859c --workdir /github/workspace 
  --rm -e "ImportOptions__ApiKeys__GitHubToken" 
  -e "ImportOptions__ApiKeys__OSPOKey" 
  -e "ImportOptions__ApiKeys__QuestKey" 
  -e "INPUT_ORG" -e "INPUT_REPO" -e "INPUT_ISSUE" 
  -e "INPUT_BRANCH" -e "HOME" -e "GITHUB_JOB" 
  -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" 
  -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" 
  -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" 
  -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" 
  -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" 
  -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" 
  -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" 
  -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" 
  -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" 
  -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" 
  -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" 
  -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" 
  -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" 
  -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" 
  -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" 
  -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_TOOL_CACHE" 
  -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" 
  -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" 
  -e GITHUB_ACTIONS=true -e CI=true 
  -v "/var/run/docker.sock":"/var/run/docker.sock" 
  -v "/home/runner/work/_temp/_github_home":"/github/home" 
  -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" 
  -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" 
  -v "/home/runner/work/docs/docs":"/github/workspace" 
  49859c:cc55a73283a34d22a3445df6d27b1f47  
  "--org" "dotnet" 
  "--repo" "dotnet/docs" 
  "--issue" "33362" 
  "--branch" "main"
```

The _Dockerfile_ itself cannot access the trailing command line args, as such, we cannot evaluate any of the options the workflow has expressed. This means that when our bash script tries to call dotnet _CleanRepo.dll_ it wouldn't know what switches to pass. As an alternative approach, I'm proposing that we try accessing the input environment variables directly.

## In this PR

1. Update _Dockerfile_ entrypoint, `ENTRYPOINT [ "/bin/sh", "/multiplexer.sh" ]`.
1. Added TODO comment about hardcoded `#if DEBUG` overriding `args`.
1. Converted the _Program.cs_ to use standalone if statements, enabling multiple switches.
1. Add _multiplexer.sh_ bash script to run the _CleanRepo.dll_ for any available switch.

> **Note**
> The naming convention for inputs in env vars is `INPUT_<INPUT-NAME>`, for more information, see [GitHub Action Workflow: Metadata syntax - Inputs](https://docs.github.com/actions/creating-actions/metadata-syntax-for-github-actions#inputs).

I think this should work, but we'll have to test it. If the bash script approach doesn't work, we should be able to easily remove the script altogether, and just call `ENTRYPOINT [ "dotnet", "/CleanRepo.dll" ]` with no other changes - assuming the _Program.cs_ changes remain in place.

Related to https://github.com/dotnet/docs/pull/34257#discussion_r1120705902

### CODE REVIEW

File: cleanrepo/CleanRepo/Program.cs [hunk 1]
```diff
@@ -24,10 +24,22 @@ class Program
     static void Main(string[] args)
     {
 #if DEBUG
+        // TODO: Consider using launchSettings.json with command line args instead:
```
Comment (BillWagner) [inline cleanrepo/CleanRepo/Program.cs:27]: With the latest changes in the script, are these necessary?

I think they're OK, but I'd like Genevieve's thoughts on any concerns with order, or running multiple commands in a single exe.

Comment (gewarren) [reply cleanrepo/CleanRepo/Program.cs:27]: The debug arguments are useful when debugging during development.

There are other changes (optimizations) that could be made if we're going to run multiple options at once - for example, only look for the docfx.json file once, or only obtain all the Markdown files once. If you want to merge it as-is, I can make those optimizations later.

Comment (IEvangelist) [reply cleanrepo/CleanRepo/Program.cs:27]: I noticed the potential for those optimizations too but didn't want to make too many changes without getting a nod of approval. Would you prefer to make those optimizations, or do you want me to?

Comment (IEvangelist) [reply cleanrepo/CleanRepo/Program.cs:27]: Also, if you prefer, I could just undo the changes to _Program.cs_ as they're not needed for this effort. Just let me know.

**Review by gewarren — APPROVED** [review]

**Pull request merged**